src/librustc/infer/error_reporting/need_type_info.rs

```diff
@@ -165,6 +175,26 @@ impl<'a, 'tcx> InferCtxt<'a, 'tcx> {
                     name,
                  ))
             }
+            Some(ty::TyS { sty: ty::Closure(def_id, substs), .. }) => {
+                let msg = " for the closure".to_string();
+                let fn_sig = substs.closure_sig(*def_id, self.tcx);
+                let args = fn_sig.inputs()
+                    .skip_binder()
+                    .iter()
+                    .next()
+                    .map(|args| args.tuple_fields()
+                        .map(|arg| arg.to_string())
+                        .collect::<Vec<_>>().join(", "))
```

Comment:
The extra indentation is _inside_ the `map`'s closure.

Reply:
Oh so it is; the `))` should move the second parenthesis to a newline tho I think (anyways I don't care much since rustfmt will eventually fix it all ^^).